api/controller/assignments.js

```diff
@@ -0,0 +1,904 @@
+    //set params
+    params.offset = offset;
+    params.limit = limit;
+    var str = [];
+    for (var p in params)
+        if (p && Object.prototype.hasOwnProperty.call(params, p)) {
+            str.push(p + "=" + params[p]);
+        }
+    return "?" + str.join("&");
+}
+
+
+function addQuery(filter, params, query, default_val) {
+    //check default case
+    query = query || {};
+    //validate
+    if (
+        typeof params[filter] != "undefined" &&
+        typeof params[filter] === "string"
+    ) {
+
+        if (filter == "name") {
+            query["name"] = {
+                $regex: new RegExp(params[filter], "i")
+            };
+        } else if (filter == "buddy_Name") {
+            
+            query["buddy_name"] = {
+                // buddy_Name is a field in the database under metadata of journal entry
+                $regex: new RegExp(params[filter], "i")
+            };
+        }
+        else if (filter == "isAssigned") {
+            query["isAssigned"] = params[filter] == "true";
+        }
+        else {
+            query[filter] = {
+                $regex: new RegExp("^" + params[filter] + "$", "i")
+            };
+        }
+
+    } else {
+        //default case
+        if (typeof default_val != "undefined") {
+            query[filter] = default_val;
+        }
+    }
+    //return
+    return query;
+}
+
+//update comment 
+exports.updateComment = function (req, res) {
+    //validate
+    if (!req.query.oid || !mongo.ObjectID.isValid(req.params.assignmentId)) {
+        return res.status(401).send({
+            'error': "Invalid assignment id",
+            'code': 35
+        });
+    }
+    var assignmentId = req.params.assignmentId;
+    var comment = JSON.parse(req.body.comment);
+    var objectId = req.query.oid;
+    db.collection(journalCollection, function (err, collection) {
+        collection.findOneAndUpdate(
+            {
+                'content.objectId': objectId,
+                'content.metadata.assignmentId': assignmentId,
+            },
+            {  //set comment only if it is match with objectId
+                $set: {
+                    'content.$[elem].metadata.comment': comment.comment,
+                }
+            },
+            {
+                safe: true,
+                arrayFilters: [{
+                    'elem.objectId': objectId
+
+                }]
+            },
+            function (err, result) {
+                if (err) {
+                    
+                    return res.status(401).send({
+                        error: "An error has occurred",
+                        code: 10
+                    });
+                }
+                if (!result) {
+                    return res.status(401).send({});
+                }
+                else {
+                    res.status(200).send(result);
+                }
+            });
+    });
+};
+
+//update status
+function updateStatus(req, res, status, objectId, callback) {
```

Comment:
`req` and `res` are not needed as parameters, they can be replaced with `assignmentId`.

Reply:
done!